warning: `QSet<T> QList<T>::toSet()` is deprecated

```shell
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/MISC/FilterableList.cpp: In member function ‘void OpenMS::Internal::FilterableList::setBlacklistItems(const QStringList&)’:
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/MISC/FilterableList.cpp:72:34: warning: ‘QSet<T> QList<T>::toSet() const [with T = QString]’ is deprecated: Use QSet<T>(list.begin(), list.end()) instead. [-Wdeprecated-declarations]
   72 |       blacklist_ = bl_items.toSet();
      |                    ~~~~~~~~~~~~~~^~
In file included from /usr/include/qt/QtCore/qhash.h:46,
                 from /usr/include/qt/QtCore/qset.h:43,
                 from /usr/include/qt/QtCore/QSet:1,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/MISC/FilterableList.h:39,
                 from /home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/MISC/FilterableList.cpp:35:
/usr/include/qt/QtCore/qlist.h:413:13: note: declared here
  413 |     QSet<T> toSet() const;
      |             ^~~~~
```

--- a/src/openms_gui/source/VISUAL/MISC/FilterableList.cpp
+++ b/src/openms_gui/source/VISUAL/MISC/FilterableList.cpp
@@ -69,20 +69,38 @@ namespace OpenMS
 
     void FilterableList::setBlacklistItems(const QStringList& bl_items)
     {
+      /*
+       * Suppressing warning toSet() deprecated till Qt 5.14
+       */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       blacklist_ = bl_items.toSet();
+#pragma GCC diagnostic pop
       updateInternalList_();
     }
 
     void FilterableList::addBlackListItems(const QStringList& items)
     {
+      /*
+       * Suppressing warning toSet() deprecated till Qt 5.14
+       */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       blacklist_.unite(items.toSet());
+#pragma GCC diagnostic pop
       updateInternalList_();
     }
 
     void FilterableList::removeBlackListItems(const QStringList& outdated_blacklist_items)
     {
       // quadratic runtime, but maintains order of items (as opposed to converting to set)
+      /*
+       * Suppressing warning toSet() deprecated till Qt 5.14
+       */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       for (const auto& bl : outdated_blacklist_items.toSet())
+#pragma GCC diagnostic pop
       {
         if (blacklist_.remove(bl) == 0)
         {

